### PR TITLE
chore: add config service and update others to latest

### PR DESCRIPTION
**Pull request by aaron-steinfeld**
## Description
This adds the config service to hypertrace-service's collection of services, and updates other services to latest. This service is needed to support hypertrace/hypertrace#142 , and will power other config-driven features in the future.

### Testing
Built this image locally and verified it in a local docker-compose deployment.

### Checklist:
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: hypertrace-service/src/main/resources/configs/hypertrace-graphql-service/default-cluster/application.conf [hunk 1]
```diff
@@ -19,4 +19,11 @@ entity.service = {
   host = ${?ENTITY_SERVICE_HOST_CONFIG}
   port = 9001
   port = ${?ENTITY_SERVICE_PORT_CONFIG}
+}
+
+config.service = {
+  host = localhost
+  host = ${?CONFIG_SERVICE_HOST_CONFIG}
```
Comment (kotharironak) [inline hypertrace-service/src/main/resources/configs/hypertrace-graphql-service/default-cluster/application.conf:26]: As a follow up, we have to modify docker-compose's ingestion component,

https://github.com/hypertrace/hypertrace/blob/main/docker/docker-compose.yml#L61
```
- CONFIG_SERVICE_HOST_CONFIG=hypertrace
- CONFIG_SERVICE_PORT_CONFIG =9001
```

File: .gitmodules [hunk 2]
```diff
@@ -22,3 +22,7 @@
 	path = config-bootstrapper
 	url = https://github.com/hypertrace/config-bootstrapper.git
 	branch = main
+[submodule "config-service"]
```
Comment (kotharironak) [inline .gitmodules:25]: I think as a follow-up, we will also need to add `config-service` as part of charts - https://github.com/hypertrace/hypertrace/blob/main/kubernetes/platform-services/Chart.yaml

**Review by kotharironak — APPROVED** [review]

**Comment by kotharironak** [comment]
@JBAhire / @aaron-steinfeld  I think, this change will also need the change in the architecture diagram - https://docs.hypertrace.org/arch/?

@JBAhire Do you think, we need any documentation for usage of this service in the context of HT at - https://github.com/hypertrace/config-service? 
- what are the usage and kind of config users can configure?

File: .gitmodules (hunk 2 above)
Comment (aaron-steinfeld) [reply .gitmodules:25]: Yep, have the changes in there ready, was waiting on this one to merge (which I suppose in retrospect wasn't necessary given that repo is using latest tag for this service)

**Comment by JBAhire** [comment]
> @JBAhire / @aaron-steinfeld I think, this change will also need the change in the architecture diagram - https://docs.hypertrace.org/arch/?
> 
> @JBAhire Do you think, we need any documentation for usage of this service in the context of HT at - https://github.com/hypertrace/config-service?
> 
>     * what are the usage and kind of config users can configure?

definitely, @kotharironak ! 

Hey @aaron-steinfeld , would you mind sending me a rough diagram for where this fits in our architecture and also RATIONALE or an issue telling why and how we are using it?

We can utilize that information to modify our architecture docs.

**Comment by kotharironak** [comment]
> > @JBAhire / @aaron-steinfeld I think, this change will also need the change in the architecture diagram - https://docs.hypertrace.org/arch/?
> > @JBAhire Do you think, we need any documentation for usage of this service in the context of HT at - https://github.com/hypertrace/config-service?
> > ```
> > * what are the usage and kind of config users can configure?
> > ```
> 
> definitely, @kotharironak !
> 
> Hey @aaron-steinfeld , would you mind sending me a rough diagram for where this fits in our architecture and also RATIONALE or an issue telling why and how we are using it?
> 
> We can utilize that information to modify our architecture docs.

@JBAhire https://github.com/hypertrace/hypertrace/issues/142

**Comment by JBAhire** [comment]
> @JBAhire [hypertrace/hypertrace#142](https://github.com/hypertrace/hypertrace/issues/142)
Yup. that I saw. I am talking specific to why we moved config service here and how it helps Hypertrace-service as a federated service moving ahead. as we have plans to convert this to macro repo as well, that will be a good input in deciding the split for repos and versioning accordingly. Basically, some info around why this is right place for config service and how it can benefit us in future while consolidation.

**Review by JBAhire — APPROVED** [review]

**Comment by aaron-steinfeld** [comment]
The config service addition is actually different than #142 (this is just the first usage of it). Here's a marked up arch diagram with changes (both related to this change and just missing links):
image

Rationale on the config service: It's a place to store configuration data such as user preferences, saved query filters, ingestion config, etc. Many of these use cases have not yet been built out. In general, this service is meant for user-managed configuration that needs to be persisted, and contains support for version history, auditing etc. In the past, we've addressed such things by spinning up individual services (such as attribute service). As new features get built out, we want to avoid that (and eventually to merge older services back into this).

This config service is designed to be a single macroservice, contained in one repo with separate modules defining different GRPC services that expose feature-specific APIs. Long term, I see our architecture having three separate logical groups - query, config and ingestion.

**Comment by JBAhire** [comment]
Thanks, @aaron-steinfeld ! This is very helpful. I will make changes in docs accordingly.

**Pull request merged**